packages/zoltan2/src/algorithms/partition/Zoltan2_AlgMultiJagged.hpp

```diff
@@ -99,9 +99,9 @@
 #define ZOLTAN2_ABS(x) ((x) >= 0 ? (x) : -(x))
 //imbalance calculation. Wreal / Wexpected - 1
 #define imbalanceOf(Wachieved, totalW, expectedRatio) \
-        double(Wachieved) / double((totalW) * (expectedRatio)) - 1
+        (Wachieved) / ((totalW) * (expectedRatio)) - 1
 #define imbalanceOf2(Wachieved, wExpected) \
-        double(Wachieved) / double(wExpected) - 1
+        (Wachieved) / (wExpected) - 1
 
 
 #define ZOLTAN2_ALGMULTIJAGGED_SWAP(a,b,temp) temp=(a);(a)=(b);(b)=temp;
```

Comment:
Please just use `std::swap` here.  If `a` or `b` are expressions with side effects, this macro will invoke the side effects multiple times.

Reply:
I'll take care of this when I merge some ongoing MJ work.